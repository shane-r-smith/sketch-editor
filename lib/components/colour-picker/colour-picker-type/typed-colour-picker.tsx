import { HexColorPicker, RgbStringColorPicker } from "react-colorful";
import type { BaseColourPickerProps } from "../colour-picker.types";
import { useCallback } from "react";
import { colord } from "colord";

import styles from "./typed-colour-picker.styles.module.scss";

export function TypedColourPicker({
  type,
  colour,
  onChange,
  ...rest
}: BaseColourPickerProps) {
  const handleChange = useCallback(
    (colour: string) => {
      onChange(colour, type);
    },
    [onChange, type]
  );

  const _className = styles.editorColourPicker;

  if (type === "RGB") {
    return (
      <RgbStringColorPicker
        className={_className}
        onChange={handleChange}
        color={colord(colour).toRgbString()}
        {...rest}
      />
    );
  }

  return (
    <HexColorPicker
      className={_className}
      color={colord(colour).toHex()}
      onChange={handleChange}
      {...rest}
    />
  );
}
