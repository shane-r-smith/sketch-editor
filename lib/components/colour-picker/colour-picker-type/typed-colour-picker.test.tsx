import { render } from "@testing-library/react";
import { TypedColourPicker } from "./typed-colour-picker";
import { vi } from "vitest";
import type { BaseColourPickerProps } from "../colour-picker.types";

describe("<TypedColourPicker />", () => {
  const defaultProps: BaseColourPickerProps = {
    type: "HEX",
    colour: "#ff0000",
    onChange: vi.fn(),
  };

  it("renders HexColorPicker for HEX type and matches snapshot", () => {
    const { container } = render(<TypedColourPicker {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it("renders RgbStringColorPicker for RGB type and matches snapshot", () => {
    const props: BaseColourPickerProps = {
      ...defaultProps,
      type: "RGB",
      colour: "#00ff00",
    };
    const { container } = render(<TypedColourPicker {...props} />);
    expect(container).toMatchSnapshot();
  });
});
