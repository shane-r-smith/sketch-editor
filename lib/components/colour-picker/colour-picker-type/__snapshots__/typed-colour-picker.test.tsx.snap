// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TypedColourPicker /> > renders HexColorPicker for HEX type and matches snapshot 1`] = `
<div>
  <div
    class="react-colorful _editorColourPicker_cffd98"
  >
    <div
      class="react-colorful__saturation"
      style="background-color: rgb(255, 0, 0);"
    >
      <div
        aria-label="Color"
        aria-valuetext="Saturation 100%, Brightness 100%"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__saturation-pointer"
          style="top: 0%; left: 100%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgb(255, 0, 0);"
          />
        </div>
      </div>
    </div>
    <div
      class="react-colorful__hue react-colorful__last-control"
    >
      <div
        aria-label="Hue"
        aria-valuemax="360"
        aria-valuemin="0"
        aria-valuenow="0"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__hue-pointer"
          style="top: 50%; left: 0%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgb(255, 0, 0);"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<TypedColourPicker /> > renders RgbStringColorPicker for RGB type and matches snapshot 1`] = `
<div>
  <div
    class="react-colorful _editorColourPicker_cffd98"
  >
    <div
      class="react-colorful__saturation"
      style="background-color: rgb(0, 255, 0);"
    >
      <div
        aria-label="Color"
        aria-valuetext="Saturation 100%, Brightness 100%"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__saturation-pointer"
          style="top: 0%; left: 100%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgb(0, 255, 0);"
          />
        </div>
      </div>
    </div>
    <div
      class="react-colorful__hue react-colorful__last-control"
    >
      <div
        aria-label="Hue"
        aria-valuemax="360"
        aria-valuemin="0"
        aria-valuenow="120"
        class="react-colorful__interactive"
        role="slider"
        tabindex="0"
      >
        <div
          class="react-colorful__pointer react-colorful__hue-pointer"
          style="top: 50%; left: 33.33333333333333%;"
        >
          <div
            class="react-colorful__pointer-fill"
            style="background-color: rgb(0, 255, 0);"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
